src/containers/BookList.js

```diff
@@ -2,40 +2,52 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import Book from '../components/Book';
-import { removeBook } from '../actions';
+import Option from '../components/Options';
+import { removeBook, changeFilter } from '../actions';
+import { catFilt } from '../components/CategoryFilter';
 
-const BookList = ({ books, handleRemove }) => (
-  <table>
-    <thead>
-      <tr>
-        <th> ID </th>
-        <th> Title </th>
-        <th> Category </th>
-      </tr>
-    </thead>
-    <tbody>
-      {books.map(({ id, title, category }) => (
-        <Book
-          key={id}
-          id={id}
-          title={title}
-          category={category}
-          clickHandler={() => handleRemove({ id, title, category })}
-        />
-      ))}
-    </tbody>
-  </table>
-);
-
-const mapStateToProps = state => {
-  const { books } = state;
-  return { books };
+const getBooksToDisplay = (state, books, filter) => {
+  switch (filter) {
+    case catFilt[0].all:
+      return books;
+    default:
+      return books.filter(book => book.category === filter);
+  }
 };
 
+const mapStateToProps = state => ({ books: getBooksToDisplay(state, state.books, state.filter) });
+
 const mapDispatchToProps = dispatch => ({
   handleRemove: book => dispatch(removeBook(book)),
+  handleFilterChange: filter => dispatch(changeFilter(filter)),
 });
 
+const BookList = ({ books, handleRemove, handleFilterChange }) => (
+  <div>
+    <Option handleFilterChange={(filter) => handleFilterChange(filter)} />
+    <table>
+      <thead>
+        <tr>
+          <th> ID </th>
+          <th> Title </th>
+          <th> Category </th>
+        </tr>
+      </thead>
+      <tbody>
+        {books.map(({ id, title, category }) => (
+          <Book
+            key={id}
+            id={id}
+            title={title}
+            category={category}
+            clickHandler={() => handleRemove({ id, title, category })}
+          />
+        ))}
+      </tbody>
+    </table>
+  </div>
+);
+
```

Comment:
- `BookList` should be the component that implements `handleFilterChange` then passes it down to `CategoryFilter` component see [screenshot]() from pathwright
- it should filter books based on `filter` value from Redux store  see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/FilterBooksBasedOnStoreFilterSelection.png) from pathwright.

You can get your `filter` value by doing this 
`const mapStateToProps = (state, ownProps) => {
  return {
    filter: state.name
  }
};`
which you used in your action